code/script.js

```diff
@@ -218,108 +224,164 @@ const generateBoard = () => {
+      charactersInPlay = charactersInPlay.filter((person) =>
+        person[category].includes(value)
+      );
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+        `No, the person is not a ${value}! Remove all people that are ${value}s.`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => !person[category].includes(value)
+      );
     }
-  } else if (category === 'other') {
-    // Similar to the one above
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the person has ${value} ${category}! Keep all people with ${value} ${category}.`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] === value
+      );
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the person doesnt have ${value} ${category}! Remove all people with ${value} ${category}.`
+      );
+      charactersInPlay = charactersInPlay.filter(
+        (person) => person[category] !== value
+      );
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
-}
+  // Invokes the function to redraw the board with the remaining people.
+  generateBoard();
+};
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// When clicking on guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  //let으로 해놓은 이유: 고르는 사람은 계속 바뀔테니까 바꿀 수 있도록 유연하게 지정
+  let personToCheck = personToConfirm;
+  // confirm 이라는 기능 자체는 boolean value: Are you sure 이라는 질문에 Yes : true, Cancel: false
+  let userConfirmed = confirm("Are you sure?"); // Displays a confirm-window
+  //userConfirmed 는 true 일 수도 있고 false도 있는 boolean value (depends on if user pushes yes or cancel)
+  if (userConfirmed) {
+    // 만일 user가 yes(take a guess) 버튼을 누르면 체크로 넘어감 (persontocheck라는 variable 가지고)
+    checkMyGuess(personToCheck); // If the player wants to guess, checkMyGuess function is invoked.
+  }
```

Comment:
Here is also a lot of comments, looks like you wrote them for yourself to understand your code which is great! A bit harder for anyone else who wants to review it to understand the comments when they are written in 2 languages. Just a thought!